typing/typecore.ml

```diff
@@ -104,7 +104,9 @@ type error =
   | Expr_type_clash of
       Errortrace.unification_error * type_forcing_context option
       * Parsetree.expression_desc option
-  | Apply_non_function of type_expr
+  | Apply_non_function of type_expr * Location.t * Location.t
+      (** Type of the function, location of extra argument,
+          location of last accepted argument. *)
```

Comment:
I think that having an inline record with documentation would be even better than having an explanation of what are each tuple elements.